.github/workflows/main.yml

```
@@ -229,46 +238,32 @@ jobs:
         env:
           DOCKER_BUILDX_OUTPUT: type=docker
         run: make docker-build-osm build-osm docker-build-tcp-echo-server
-      # PR Tests
-      - name: Run PR tests
-        id: pr_test
-        if: ${{ github.event_name == 'pull_request' }}
+      - name: Run tests
+        id: test
         env:
           K8S_NAMESPACE: "osm-system"
-        run: go test ./tests/e2e -test.v -ginkgo.v -ginkgo.progress -installType=KindCluster -test.timeout 0 -test.failfast -ginkgo.failFast -ginkgo.focus='\[Bucket ${{ matrix.bucket }}\]'
+        run: go test ./tests/e2e -test.v -ginkgo.v -ginkgo.progress -installType=KindCluster -kindClusterVersion='${{ matrix.k8s_version }}' -test.timeout 0 -test.failfast -ginkgo.failFast -ginkgo.focus='\[Bucket ${{ matrix.bucket }}\].*${{ matrix.focus }}'
         continue-on-error: true
-      - name: Upload PR test logs
-        if: ${{ steps.pr_test.conclusion != 'skipped' }}
-        uses: actions/upload-artifact@v2
-        with:
-          name: pr_test_logs_bucket_${{ matrix.bucket }}
-          path: /tmp/test**/*
-      - name: Check continue PR tests
-        if: ${{ steps.pr_test.conclusion != 'skipped' && steps.pr_test.outcome == 'failure'}}
-        run: exit 1
-      - name: Clean PR tests
-        if: ${{ steps.pr_test.conclusion != 'skipped' }}
-        run: rm -rf /tmp/test*
-
-      # Push Tests
-      - name: Run Push tests
-        id: push_test
-        if: ${{ github.event_name == 'push' }}
-        env:
-          K8S_NAMESPACE: "osm-system"
-        run: go test ./tests/e2e -test.v -ginkgo.v -ginkgo.progress -installType=KindCluster -test.timeout 0 -test.failfast -ginkgo.failFast -ginkgo.focus='\[Bucket ${{ matrix.bucket }}\]'
-        continue-on-error: true
-      - name: Upload Push test logs
-        if: ${{ steps.push_test.conclusion != 'skipped' }}
+      - name: Set Logs name
+        if: ${{ steps.test.conclusion != 'skipped' }}
+        id: logname
+        run: |
+          if [[ -n "${{ matrix.k8s_version }}" ]]; then
+            echo "::set-output name=name::test_logs_k8s_version_${{ matrix.k8s_version }}"
+          else
+            echo "::set-output name=name::test_logs_bucket_${{ matrix.bucket }}"
```

```suggestion
            echo "::set-output name=BUCKET::test_logs_bucket_${{ matrix.bucket }}"
```

Same as above, rename `name` to something more descriptive.